### ISSUE TITLE
Preserving total intensity

### ISSUE THREAD

**Issue by george-tog** [WeisongZhao/sparse-deconv-py#4]
Hello Dr. Zhao,

I am currently applying sparse deconv to neuron imaging, and I am interested in modifying the loss function to include a term for total intensity conservation, or ideally, local intensity conservation, in hopes of retaining quantitative intensity information. My basic idea is to include a term that is the squared difference of total intensity in the reconstruction and original images. I think using a reasonably large hyperparameter constant would enforce the conservation of intensity while still leveraging sparsity/continuity priors.

However, I am not familiar enough with the optimization method used to understand how to include such a term in the Fourier update equations. Do you think this idea is reasonable, and if so, could you help me with the update equations? (even if it's just showing the mathematical form, I can manage the implementation)

Best,
George

**Comment by WeisongZhao** [comment]
Hi George,

Thank you for your interests. Brilliant idea! Actually, the image fidelity term (Ax-y) will do this "intensity conservation" for you. If you want to apply `Sparse deconvolution` on neuron imaging, please turn off the background estimation feature.
It is maybe relevant to our [Extended Data Fig. 10](https://www.nature.com/articles/s41587-021-01092-2/figures/16). (when we processed Ca2+ imaging data in Extended Data Fig. 10, we selected `no background estimation` option to avoid removing the baseline signal.)
